### ISSUE TITLE
Setting client presence invalidates session

### ISSUE THREAD

**Issue by BannerBomb** [discordjs/discord.js#5316]
**Please describe the problem you are having in as much detail as possible:**
Since #4879 was merged which changed the setPresence data "activity" key to "activities"
1. if you set the client status twice during startup at least one time calling with invalid data will cause the session to invalidate. Which is related to what happens in my #4896 issue. (I forgot that this was mentioned here https://github.com/discord/discord-api-docs/issues/2180#issuecomment-754672069 before I made this issue)
2. The PresenceData interface in the typings also hasn't been updated for the new key.

**Include a reproducible code sample here, if possible:**
```js
const { Client } = require('discord.js');
const client = new Client({
    intents: [...],
    presence: { // This presence has the invalid data in this example. Since it is still using `activity`
        activity: {
            name: '...Powering up',
            type: 0
        }
    }
});

client.on('ready', () => {
    client.user.setPresence({ // This will now invalidate the session
        status: 'online',
        activities: [{
            name: 'Test',
            type: 'WATCHING'
        }]
    })
});

client.login(...);
```
```js
const { Client } = require('discord.js');
const client = new Client({
    intents: [...],
    presence: { // This presence has the valid data in this example. Since it is using the new  `activities` key.
        activities: [{
            name: '...Powering up',
            type: 0
        }]
    }
});

client.on('ready', () => {
    client.user.setPresence({ // This will now invalidate the session
        status: 'online',
        activity: {
            name: 'Test',
            type: 'WATCHING'
        }
    });
});

client.login(...);
```
```js
const { Client } = require('discord.js');
const client = new Client({
    intents: [...],
    presence: { // This presence has invalid data.
        activity: {
            name: '...Powering up',
            type: 0
        }
    }
});

client.on('ready', () => {
    client.user.setPresence({ // This will now invalidate the session
        status: 'online',
        activity: {
            name: 'Test',
            type: 'WATCHING'
        }
    })
});

client.login(...);
```

However, setting the presence only once with invalid data won't invalidate the session.
```js
const { Client } = require('discord.js');
const client = new Client({
    intents: [...],
    presence: { // This presence has invalid data.
        activity: {
            name: '...Powering up',
            type: 0
        }
    }
});

client.on('ready', () => {
    // I removed the setPresence method from here.
})

client.login(...);
```

When the session is invalidated this is what the debug event shows.
```
[2021-02-12 02:00:25] [WS => Shard 0] [INVALID SESSION] Resumable: false.
[2021-02-12 02:00:25] [SHARD RECONNECTING] Shard 0
[2021-02-12 02:10:20] [WS => Shard 0] [CLOSE]
[2021-02-12 02:10:20]     Event Code: 4003
[2021-02-12 02:10:20]     Clean     : true
[2021-02-12 02:10:20]     Reason    : Not authenticated.
[2021-02-12 02:10:20] [WS => Shard 0] Clearing the heartbeat interval.
[2021-02-12 02:10:20] [SHARD RECONNECTING] Shard 0
[2021-02-12 02:10:20] [WS => Shard 0] WS State: CLOSED
```

**Further details:**

- discord.js version: v12.5.0 (master)
- Node.js version: v15.7.0
- Operating system: Ubuntu 18.04.2 LTS
- Priority this issue should have – please be realistic and elaborate if possible: low

**Relevant client options:**

- partials: none
- gateway intents: `[Intents.FLAGS.GUILD_MEMBERS, Intents.FLAGS.GUILDS, Intents.FLAGS.GUILD_MESSAGES, Intents.FLAGS.GUILD_MESSAGE_REACTIONS, Intents.FLAGS.GUILD_VOICE_STATES]`
- other: none

- I have also tested the issue on latest master, commit hash: `ee5bc1a`

**Comment by pettinen** [comment]
I think I found a related problem. (Should it warrant its own issue? Not sure. #5317 doesn't currently seem to fix it though.)

Bots seem to disconnect with the `[WS => Shard 0] [INVALID SESSION] Resumable: false.` message also when `activities` is `null` (even though the [Gateway](https://discord.com/developers/docs/topics/gateway#update-status-gateway-status-update-structure) claims to allow that).

This might happen e.g. when clearing activity with `client.user.setActivity();`

Setting `activities` to an empty array instead of `null` seems to fix this:
```
node_modules/discord.js/src/structures/ClientPresence.js
@@ -37,5 +37,5 @@
     };
     if (activities === null) {
-      data.activities = null;
+      data.activities = [];
       return data;
     }
```

Tested on master (d744e51).
